low-level/framework/utils/iem.py

```diff
@@ -155,6 +153,20 @@ def check_exsisting_fault_iems(self):
             if prev_fault_iem_event:
                 self.fault_iems.append(event_name)
 
-def generate_iem(iem):
-    """Generate iem."""
-    syslog.syslog(iem)
+    @staticmethod
+    def generate_iem(module, event_code, severity, description):
+        """Generate iem and send it to a MessgaeBroker."""
+        try:
+            logger.info(f"Sending IEM alert for module:{module}"
+                        f" and event_code:{event_code}")
+            # check if IEM Framework initialized,
+            # if not, retry initializing the IEM Frameowork
+            if os.path.exists(IEM_INIT_FILE):
```

Comment:
We need to link current sspl pid also in this file. Consider init had failed with some previous sspl service run and IEM were never generated as no situation arises or say this file is not cleaned up by running old sspl instance.
And then new SSPL service instance started, for which IEM init was successful actually, still if this file is present, init would be tried, which is not the intention.

Better you refer and link using SSPL pid, like recently we had done for support bundle case, check #425

Reply:
Ack.
done the changes